### PR TITLE
BSP-289: refactoring

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BSP-289

### Change description ###
Sending bulk print letter on HWF success 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsController.java [hunk 1]
```diff
@@ -38,12 +46,16 @@
         validateCaseData(callbackRequest);
         Map<String, Object> caseData = callbackRequest.getCaseDetails().getData();
         if (isConsentedApplication(caseData)) {
-            if (isSolicitorAgreedToReceiveEmails(caseData, "solicitorAgreeToReceiveEmails")) {
-                log.info("Sending Consented HWF Successful email notification to Solicitor");
-                notificationService.sendHWFSuccessfulConfirmationEmail(callbackRequest);
+            if (isSolicitorAgreedToReceiveEmails(caseData, SOLICITOR_AGREE_TO_RECEIVE_EMAILS)) {
+                if (isPaperApplication(caseData)) {
+                    log.info("Sending Consented HWF Successful letter to bulk print");
+                    bulkPrintService.sendForBulkPrint(caseDocumentForHwfSuccess(), callbackRequest.getCaseDetails());
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsController.java:52]: this is going to be different service

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java [hunk 2]
```diff
@@ -13,28 +13,74 @@
...
+
+            caseData.put(BULK_PRINT_COVER_SHEET_APP, applicantCoverSheet);
+            caseData.put(BULK_PRINT_LETTER_ID_APP, applicantLetterId);
+
+            logInfo("Applicant", applicantCoverSheet, applicantLetterId);
+
+            return applicantLetterId;
+        }
+
+        return UUID.fromString("");
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java:64]: I have some other ideas how to do it, but this would be done in another PR

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintServiceTest.java [hunk 3]
```diff
@@ -41,39 +45,40 @@ public void setUp() {
...
         assertThat(bulkPrintRequestArgumentCaptor.getValue().getBulkPrintDocuments().size(), is(2));
     }
 
-    private CaseDetails caseDetails() throws Exception {
-        try (InputStream resourceAsStream =
-                 getClass().getResourceAsStream("/fixtures/bulk-print.json")) {
-            return mapper.readValue(resourceAsStream, CallbackRequest.class).getCaseDetails();
-        }
-    }
+    //    private CaseDetails caseDetails() throws Exception {
```
Comment (yarpo) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintServiceTest.java:76]: I just wanted to make it green. I will write more unit tests when I finish refactoring

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsController.java [hunk 4]
```diff
@@ -27,23 +33,31 @@
...
         validateCaseData(callbackRequest);
         Map<String, Object> caseData = callbackRequest.getCaseDetails().getData();
         if (isConsentedApplication(caseData)) {
-            if (isSolicitorAgreedToReceiveEmails(caseData, "solicitorAgreeToReceiveEmails")) {
-                log.info("Sending Consented HWF Successful email notification to Solicitor");
-                notificationService.sendHWFSuccessfulConfirmationEmail(callbackRequest);
+            if (isSolicitorAgreedToReceiveEmails(caseData, SOLICITOR_AGREE_TO_RECEIVE_EMAILS)) {
+                if (isPaperApplication(caseData)) {
+                    log.info("Sending Consented HWF Successful letter to bulk print");
+                    bulkPrintService.sendLetterToApplicantSolicitor(authToken, callbackRequest.getCaseDetails());
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsController.java:54]: this is still a stub. No logic yet. Todo in next step.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/BulkPrintControllerTest.java [hunk 5]
```diff
@@ -1,164 +1,76 @@
...
-    private static final String CONTESTED_BULK_PRINT_SIMPLE_JSON = "/fixtures/contested/bulk_print_simple.json";
-    private static final String CONTESTED_BULK_PRINT_CONSENT_ORDER_APPROVED_JSON
-        = "/fixtures/contested/bulk_print_consent_order_approved.json";
-    private static final String CONTESTED_BULK_PRINT_CONSENT_ORDER_NOT_APPROVED_JSON
-        = "/fixtures/contested/bulk_print_consent_order_not_approved.json";
-
-    private final ObjectMapper objectMapper = new ObjectMapper();
+    private static final String BULK_PRINT_TO_APPLICANT_AND_RESPONDENT
+            = "fixtures/contested/bulk_print_consent_order_not_approved.json";
+    private static final String BULK_PRINT_TO__RESPONDENT_ONLY
```
Comment (jvictorcf) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/BulkPrintControllerTest.java:33]: I think there's an extra underscore here.

**Comment by yarpo** [comment]
It's too old to maintain. Closing off.

A lot of things has been done already in other PRs
